File: model/genimage.py
from google import genai
from google.genai import types
from PIL import Image
from io import BytesIO
import os

class GeminiImageGenerator:
    def __init__(self, api_key: str, model: str = "gemini-2.0-flash-exp-image-generation"):
        self.client = genai.Client(api_key=api_key)
        self.model = model

    def generate_image(self, prompt: str, save_path: str = "gemini_image.png") -> str:
        response = self.client.models.generate_content(
            model=self.model,
            contents=prompt,
            config=types.GenerateContentConfig(response_modalities=['Text', 'Image'])
        )

        image_saved = False
        for part in response.candidates[0].content.parts:
            if part.text is not None:
                print("Text Response:\n", part.text)
            elif part.inline_data is not None:
                image = Image.open(BytesIO(part.inline_data.data))
                image.save(save_path)
                image.show()
                image_saved = True

        if image_saved:
            return save_path
        return "No image generated."

# Example usage:
if __name__ == "__main__":
    api_key = ""
    prompt = (
        "Create a highly detailed cinematic scene showing Iron Man and Loki teaming up to save the multiverse. "
        "Include the multiverse tree, a fierce battle with Kang variants, and Doctor Doom emerging from shadows."
    )
    generator = GeminiImageGenerator(api_key)
    image_path = generator.generate_image(prompt)
    print("Image saved at:", image_path)

File: model/genimage.js
import fs from "fs";
import { GoogleGenerativeAI } from "@google/generative-ai";

// Optional: Only needed if you want to log base64 images as actual previews
// import { createCanvas, loadImage } from "canvas";

class GeminiImageGenerator {
  constructor(apiKey, model = "gemini-2.0-flash-exp-image-generation") {
    this.apiKey = apiKey;
    this.modelName = model;
    this.client = new GoogleGenerativeAI(apiKey);
  }

  async generateImage(prompt, savePath = "gemini_image.png") {
    const model = this.client.getGenerativeModel({ model: this.modelName });

    const result = await model.generateContent({
      contents: [{ role: "user", parts: [{ text: prompt }] }],
      generationConfig: {
        response_mime_type: ["image/png", "text/plain"],
      },
    });

    let imageSaved = false;

    for (const part of result.response.parts) {
      if (part.text) {
        console.log("Text Response:\n", part.text);
      } else if (part.inlineData && part.inlineData.mimeType.startsWith("image/")) {
        const buffer = Buffer.from(part.inlineData.data, "base64");
        fs.writeFileSync(savePath, buffer);
        console.log("Image saved at:", savePath);
        imageSaved = true;
      }
    }

    return imageSaved ? savePath : "No image generated.";
  }
}

// Example usage
const main = async () => {
  const apiKey = ""; // 🔐 Add your Gemini API key here

  const prompt = `
    Create a highly detailed cinematic scene showing Iron Man and Loki teaming up to save the multiverse. 
    Include the multiverse tree, a fierce battle with Kang variants, and Doctor Doom emerging from shadows.
  `;

  const generator = new GeminiImageGenerator(apiKey);
  const imagePath = await generator.generateImage(prompt);
  console.log("Final Image Path:", imagePath);
};

main();
